packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -333,15 +354,24 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection {
 
         this.socket.on("get_ops_response", (result) => {
             const messages = result.messages as ISequencedDocumentMessage[];
-            if ((result.nonce as string).startsWith(this.requestOpsNoncePrefix) && messages.length > 0) {
-                this.logger.sendTelemetryEvent({
-                    eventName: "GetOps",
-                    first: messages[0].sequenceNumber,
-                    last: messages[messages.length - 1].sequenceNumber,
-                    code: result.code,
-                    length: messages.length,
-                });
-                this.socket.emit("op", this.documentId, messages);
+            const data = this.getOpsMap.get(result.nonce);
+            // Due to socket multiplexing, this client may not have asked for any data
+            // If so, there it most likely does not need these ops (otherwise it already asked for them)
+            if (data !== undefined) {
+                this.getOpsMap.delete(result.nonce);
+                if (messages.length > 0) {
```

Comment:
I missed this in the first pr. `messages` could be `undefined` in error cases.
So typing should be `const messages = result.messages as (ISequencedDocumentMessage[] | undefined);`
Can you switch the check to `if (messages && messages.length > 0)`

Reply:
Done